core/src/test/java/org/apache/calcite/test/TableInRootSchemaTest.java

```diff
@@ -97,7 +102,8 @@
     private Class[] columnTypes = { String.class, Integer.class };
     private Object[][] rows = new Object[3][];
 
-    SimpleTable() {
+    // CHECKSTYLE: IGNORE 1
```

Comment:
I wonder what changed; we were using this in tests before.

I see you made `TableInRootSchemaTest` public. Maybe so that you could reference it from another test? I know that SimpleTable is referenced from a few tests already, but it's getting to be too much coupling. Better to move the sub-class into `Smalls`, or something.

I also see that you are calling `Expressions.new_(Constructor)`. Maybe if you instead called `Expressions.new_(Type)` you would not need a public default constructor.

Reply:
In the new test, a `SimpleTable` table is returned by the table function, so it may actually not be present in the current schema. To simulate it, I've overridden `SimpleTable.getExpression()` to use table constructor directly, since the default implementation obtains it by name from the schema. So it was the change that required changing the modifier.

I've moved `SimpleTable` to the `Smalls` class, also I've removed the suppression since now it is referenced from different packages, and the public modifier is legal. 
Thanks for referring to the method that accepts type, I've used it now.